staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/dispatcher_test.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mutating
+
+import (
+	"encoding/json"
+	"testing"
+)
+
+func TestMutationAnnotationValue(t *testing.T) {
+	tcs := []struct {
+		config   string
+		webhook  string
+		mutated  bool
+		expected string
+	}{
+		{
+			config:   "test-config",
+			webhook:  "test-webhook",
+			mutated:  true,
+			expected: `{"configuration": "test-config", "webhook": "test-webhook", "mutated": "true"}`,
+		},
+		{
+			config:   "test-config",
+			webhook:  "test-webhook",
+			mutated:  false,
+			expected: `{"configuration": "test-config", "webhook": "test-webhook", "mutated": "false"}`,
+		},
+	}
+
+	for _, tc := range tcs {
+		actual := mutationAnnotationValue(tc.config, tc.webhook, tc.mutated)
+		if actual != tc.expected {
+			t.Errorf("composed mutation annotation value doesn't match, want: %s, got: %s", tc.expected, actual)
+		}
+	}
+}
+
+func TestPatchAnnotationValue(t *testing.T) {
+	tcs := []struct {
+		config   string
+		webhook  string
+		patch    string
+		expected string
+	}{
+		{
+			// valid patch annotation
+			config:   "test-config",
+			webhook:  "test-webhook",
+			patch:    `[{"op": "add", "path": "/metadata/labels/a", "value": "true"}]`,
+			expected: `{"configuration": "test-config", "webhook": "test-webhook", "patch": "[{\"op\": \"add\", \"path\": \"/metadata/labels/a\", \"value\": \"true\"}]"}`,
```

Comment:
Would it make sense to inline the patch json instead of escaping it and putting it in a string?

In the future we might support other patch formats. Also include the patch type field to future proof this?

Reply:
good point. Added patch type field in the payload to future proof other patch formats. Added code to ensure the patch binary is a valid JSON patch and then inline it